addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRule.java

```diff
@@ -304,57 +254,109 @@ public void scan() {
             // only in case of a bare XML parser which execute the conetnt and then
             // gives it back almost untouched (maybe because it applies an XSLT or
             // query it using XPath and give back the result).
-            msg = getNewMsg();
+            localFileInclusionAttack(getNewMsg());
+        }
+    }
 
-            try {
-                String localFile;
-                String response;
-                Matcher matcher;
-
-                for (int idx = 0; idx < LOCAL_FILE_TARGETS.length; idx++) {
-                    // Prepare the message
-                    localFile = LOCAL_FILE_TARGETS[idx];
-                    payload = MessageFormat.format(ATTACK_HEADER + ATTACK_BODY, localFile);
-                    // msg = getNewMsg();
-                    msg.setRequestBody(payload);
-
-                    // Send message with local file inclusion
-                    sendAndReceive(msg);
-
-                    // Parse the result
-                    if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.OK) {
-
-                        response = msg.getResponseBody().toString();
-                        matcher = LOCAL_FILE_PATTERNS[idx].matcher(response);
-                        if (matcher.find()) {
-
-                            newAlert()
-                                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
-                                    .setAttack(payload)
-                                    .setEvidence(matcher.group())
-                                    .setMessage(msg)
-                                    .raise();
-                        }
-                    }
-
-                    // Check if the scan has been stopped
-                    // if yes dispose resources and exit
-                    if (isStop()) {
-                        // Dispose all resources
-                        // Exit the rule
-                        return;
-                    }
+    // TODO: Add Doc Comments
+    protected void localFileReflectionAttack(HttpMessage msg) {
```

Comment:
Can be private. Will explain further while commenting on the other class.

Reply:
This is private now, as the tests are using the `scan()` method.